targets/nuttx-stm32f4/jerry_main.c

```diff
@@ -447,58 +384,66 @@ int jerry_main (int argc, char *argv[])
     }
   }
 
-  if (files_counter == 0)
-  {
-    printf ("No input files, running a hello world demo:\n");
-    char *source_p = "var a = 3.5; print('Hello world ' + (a + 1.5) + ' times from JerryScript')";
-
-    jerry_run_simple ((jerry_char_t *) source_p, strlen (source_p), flags);
-    return JERRY_STANDALONE_EXIT_CODE_OK;
-  }
-
   jerry_init (flags);
 
-  register_js_function ("assert", assert_handler);
-  register_js_function ("gc", gc_handler);
+  jerryx_handler_register_global ("assert", jerryx_handler_assert);
+  jerryx_handler_register_global ("gc", jerryx_handler_gc);
+  jerryx_handler_register_global ("print", jerryx_handler_print);
```

Comment:
the value returned by `jerryx_handler_register_global` must be freed by `jerry_release_value` (that's why jerry-main has kept `register_js_function` as a wrapper around `jerryx_handler_register_global`, to ensure that such extra steps are performed, see https://github.com/jerryscript-project/jerryscript/blob/master/jerry-main/main-unix.c#L313)